### PR TITLE
fix: input types loose out on provided graphQL documentation

### LINKED ISSUE #1625: bug(config): input types loose out on provided graphQL documentation
Documentation isn't copied into the blueprint for input types from configurations. For eg:

```graphql
"""
Some Documentation
"""
input Foo {
  id: Int!
}
```

While using a configuration with an input type like the above, documentation isn't copied in the final schema. 

**Technical Requirements**
- Copy the documentation into the blueprint
- Ensure integration tests are added

### PR DESCRIPTION
**Pull request by sn99**
**Summary:**  
Add descriptions to input object types

**Issue Reference(s):**  
Fixes #1625

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

/claim #1625

## Summary by CodeRabbit

- **New Features**
	- Enhanced parsing capabilities for SDL descriptions, improving the creation of types with documentation and fields.
- **Tests**
	- Added a new test for verifying the improved parsing functionality of SDL descriptions.

### CODE REVIEW

File: src/config/config.rs [hunk 1]
```diff
@@ -835,4 +835,30 @@ mod tests {
...
+        let expected = Config::default().types(vec![(
+            "Foo",
+            Type {
+                doc: Some("Some Documentation".to_string()),
+                ..Default::default()
+            }
+            .fields(vec![("id", Field::int().required(true))]),
+        )]);
+        assert_eq!(actual, expected);
+    }
```
[the remark this answers is not in the document]
Comment (tusharmath) [reply src/config/config.rs:863]: You can drop this test and add one like this https://github.com/tailcallhq/tailcall/tree/main/tests/execution

[the remark this answers is not in the document]
Comment (sn99) [reply src/config/config.rs:863]: @tusharmath I am having trouble with what the test will look like, can you give me a hint?

[the remark this answers is not in the document]
Comment (tusharmath) [reply src/config/config.rs:863]: Check out this test — `tests/execution/test-query-documentation.md`

[the remark this answers is not in the document]
Comment (sn99) [reply src/config/config.rs:863]: @tusharmath does the test look alright?

**Pull request closed without merging**